lib/FrontendTool/FrontendTool.cpp

```diff
@@ -424,6 +424,15 @@ static bool buildModuleFromInterface(CompilerInstance &Instance) {
   StringRef ABIPath = Instance.getPrimarySpecificPathsForAtMostOnePrimary()
                           .SupplementaryOutputs.ABIDescriptorOutputPath;
 
+  bool IgnoreAdjacentModules = false;
+  if (Invocation.getFrontendOptions().RequestedAction !=
+        FrontendOptions::ActionType::CompileModuleFromInterface &&
+      Invocation.getFrontendOptions().RequestedAction !=
+        FrontendOptions::ActionType::TypecheckModuleFromInterface &&
+      !Invocation.getFrontendOptions().ExplicitInterfaceBuild) {
+      IgnoreAdjacentModules = Instance.getASTContext().IgnoreAdjacentModules;
```

Comment:
These conditions are a bit opaque to me - could we have a comment explaining why we don't honour the ASTContext flag in these cases?

Reply:
Those are cases when the outer compiler instance doesn't instantiate an ASTContext. I replaced this check with the more direct check of `Instance.hasASTContext`.

Reply:
Nice, that makes it obvious what's going on. Thanks!